src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Renamer.cs

```diff
@@ -77,15 +77,17 @@ internal sealed class Renamer
 
         public Task RenameAsync(Project project)
         {
-            return GetStrategy()?.RenameAsync(project, _oldFilePath, _newFilePath) ?? Task.CompletedTask;
+            return GetStrategyAsync(project).Result?.RenameAsync(project, _oldFilePath, _newFilePath) ?? Task.CompletedTask;
```

Comment:
use await, not `.Result`.

Additionally, you should do a null check as GetStrategyAsync can return null result.